### PR TITLE
Assessent complete

### PR DESCRIPTION
**Pull request by ninerboy81**
# Submitting Week 4 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [ ✅] JavaScript Coding Challenges
- [✅ ] Ruby Coding Challenges
- [✅ ] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -20,9 +20,57 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
...
+        const actualShuffle = shuffleArr(colors1)
+        expect(actualShuffle.sort()).toEqual(colors1.sort())
+    })
+    it("returns shuffled array when called with an array", () => { 
+        const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+        const actualShuffle = shuffleArr(colors2)
+        expect(actualShuffle.sort()).toEqual(colors2.sort())
+    })
+    
+})
```
Comment (elysemontano) [inline code-challenges.test.js:35]: You should only need one it method used in your test. You will have multiple expect statements for each test variable, but no need to write multiple it statements. You will also want to avoid performing any logic like sort in your test. The expect method should invoke the method and pass in your test variables and compare a set expected output.

File: code-challenges.test.js [hunk 2]
```diff
@@ -20,9 +20,57 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
...
+      i = Math.floor(Math.random() * m--)
+  
+      // here the array will swap the data that was shuffled with two separate arrays
+      t = array[m]
+      array[m] = array[i]
+      array[i] = t
+    }
+  
+    return array
+  }
```
Comment (elysemontano) [inline code-challenges.test.js:72]: Pretty cool approach

File: code-challenges.test.js [hunk 3]
```diff
@@ -45,5 +127,29 @@ const testArray1 = [3, 7, 10, 5, 4, 3, 3]
...
+// 2. create a variable that will store the no duplicate values
+// 3. use the spread operator to flatten the arrays
+// 4. use the set method to remove duplicate values
+// 5. return the array
+// 6. return the no duplicate values
+
 
 // b) Create the function that makes the test pass.
+
+const noDupes = (...arrays) => {
```
Comment (elysemontano) [inline code-challenges.test.js:152]: Nice!!  Great work on implementing the stretch challenge!

File: code-challenges.test.js [hunk 4]
```diff
@@ -33,9 +81,43 @@ const nums1 = [3, 56, 90, -8, 0, 23, 6]
 const nums2 = [109, 5, 9, 67, 8, 24]
 // Expected output: [5, 109]
 
+describe ("when minMax is called", () => {
+    it("returns an array of the minimum and maximum numbers in that order", () => {
+        const actualMinMax = minMax(nums1)
+        const actualMinMax2 = minMax(nums2)
```
Comment (elysemontano) [inline code-challenges.test.js:87]: This is an interesting approach, but it certainly does what it is supposed to. You can also invoke this inside the expect method which may be a little more concise.

File: interview-questions.md [hunk 5]
```diff
@@ -7,52 +7,52 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 3. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: From what i can remember is that object oriented programming mean that most if not all your code that you write will be created with objects. This means that you can create objects that have properites and methods. This is different than functional programming, which is when you create functions that are not objects. 
 
-  Researched answer:
+  Researched answer: Functional programming is a top down stragety meaning that you create functions that are not objects in a hierarchy. Object oriented programming is a bottom up strategy meaning that you create objects that have properties and methods. You also focus more attention on the data that you are creating rather than the functions that you are creating.
```
Comment (elysemontano) [inline interview-questions.md:28]: Nice definition!

File: ruby-challenges.rb [hunk 6]
```diff
@@ -31,3 +52,17 @@
...
+    if string.downcase == string.downcase.reverse
+      "#{string} is a palindrome"
+    else
+      "#{string} is not a palindrome"
+    end
+end
+
+p palindrome?(palindrome_tester1)
+p palindrome?(palindrome_tester2)
+p palindrome?(palindrome_tester3)
```
Comment (elysemontano) [inline ruby-challenges.rb:68]: Nice work!  Crushed this section!  If you want to stretch yourself, consider edge cases when applicable.

**Review by elysemontano — COMMENTED** [review]
Great work on this assessment Luis!  It looks like your first test might have gotten a little thrown off, but other than that your code for both Javascript and Ruby look great and your interview answers are really well done.  I also really appreciate your participation in class when we go through the assessments.  Keep up the great work, and make sure to resist the urge to merge until I have had the chance to review and approve.
